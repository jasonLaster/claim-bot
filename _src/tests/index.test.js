const { getPayload, tests } = require("./helpers");

function getComment(body) {
  const comment = require(`./fixtures/issue_comment.created.json`);
  comment.payload.comment.body = body;
  return comment;
}

const APIS = {
  issue_comment: {
    created: [
      "issues.addLabels",
      "issues.removeLabel",
      "issues.removeAssigneesFromIssue",
      "issues.addAssigneesToIssue",
      "issues.createComment",
      "orgs.getTeamMembership",
      "orgs.addTeamMembership"
    ]
  },
  issue: {
    closed: ["issues.removeLabel"],
    opened: ["issues.addLabels"]
  }
};

const opened = getPayload("issue.opened");
const closed = getPayload("issue.closed");
const claimed = getComment("/claim");
const unclaimed = getComment("/unclaim");

tests("Claim Bot", { APIS, log: false }, [
  {
    name: "issue closed",
    actions: ["issue.closed"],
    events: [closed]
  },
  {
    name: "issue opened",
    actions: ["issue.opened"],
    events: [opened]
  },
  {
    name: "claimed",
    actions: ["issue_comment.created"],
    events: [claimed]
  },
  {
    name: "claimed, unclaimed",
    actions: ["issue_comment.created"],
    events: [claimed, unclaimed]
  },
  {
    name: "opened, claimed",
    actions: ["issue.opened", "issue_comment.created"],
    events: [opened, unclaimed]
  }
]);