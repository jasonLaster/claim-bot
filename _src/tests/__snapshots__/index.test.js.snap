// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Claim Bot claimed 1`] = `
Object {
  "issues": Object {
    "addAssigneesToIssue": Array [],
    "addLabels": Array [],
    "createComment": Array [
      Array [
        Object {
          "body": "Done :+1:
",
          "number": 4501,
          "owner": "devtools-html",
          "repo": "debugger.html",
        },
      ],
    ],
    "removeAssigneesFromIssue": Array [
      Array [
        Object {
          "body": Object {
            "assignees": Array [
              "jasonLaster",
            ],
          },
          "number": 4501,
          "owner": "devtools-html",
          "repo": "debugger.html",
        },
      ],
    ],
    "removeLabel": Array [
      Array [
        Object {
          "name": "in progress",
          "number": 4501,
          "owner": "devtools-html",
          "repo": "debugger.html",
        },
      ],
    ],
  },
  "orgs": Object {
    "addTeamMembership": Array [],
    "getTeamMembership": Array [],
  },
}
`;

exports[`Claim Bot claimed, unclaimed 1`] = `
Object {
  "issues": Object {
    "addAssigneesToIssue": Array [],
    "addLabels": Array [],
    "createComment": Array [
      Array [
        Object {
          "body": "Done :+1:
",
          "number": 4501,
          "owner": "devtools-html",
          "repo": "debugger.html",
        },
      ],
      Array [
        Object {
          "body": "Done :+1:
",
          "number": 4501,
          "owner": "devtools-html",
          "repo": "debugger.html",
        },
      ],
    ],
    "removeAssigneesFromIssue": Array [
      Array [
        Object {
          "body": Object {
            "assignees": Array [
              "jasonLaster",
            ],
          },
          "number": 4501,
          "owner": "devtools-html",
          "repo": "debugger.html",
        },
      ],
      Array [
        Object {
          "body": Object {
            "assignees": Array [
              "jasonLaster",
            ],
          },
          "number": 4501,
          "owner": "devtools-html",
          "repo": "debugger.html",
        },
      ],
    ],
    "removeLabel": Array [
      Array [
        Object {
          "name": "in progress",
          "number": 4501,
          "owner": "devtools-html",
          "repo": "debugger.html",
        },
      ],
      Array [
        Object {
          "name": "in progress",
          "number": 4501,
          "owner": "devtools-html",
          "repo": "debugger.html",
        },
      ],
    ],
  },
  "orgs": Object {
    "addTeamMembership": Array [],
    "getTeamMembership": Array [],
  },
}
`;

exports[`Claim Bot issue closed 1`] = `
Object {
  "issues": Object {
    "removeLabel": Array [
      Array [
        Object {
          "name": "in progress",
          "number": 4384,
          "owner": "devtools-html",
          "repo": "debugger.html",
        },
      ],
      Array [
        Object {
          "name": "available",
          "number": 4384,
          "owner": "devtools-html",
          "repo": "debugger.html",
        },
      ],
    ],
  },
}
`;

exports[`Claim Bot issue opened 1`] = `
Object {
  "issues": Object {
    "addLabels": Array [
      Array [
        Object {
          "labels": Array [
            "not-available",
          ],
          "number": 4501,
          "owner": "devtools-html",
          "repo": "debugger.html",
        },
      ],
    ],
  },
}
`;

exports[`Claim Bot opened, claimed 1`] = `
Object {
  "issues": Object {
    "addAssigneesToIssue": Array [],
    "addLabels": Array [
      Array [
        Object {
          "labels": Array [
            "not-available",
          ],
          "number": 4501,
          "owner": "devtools-html",
          "repo": "debugger.html",
        },
      ],
    ],
    "createComment": Array [
      Array [
        Object {
          "body": "Done :+1:
",
          "number": 4501,
          "owner": "devtools-html",
          "repo": "debugger.html",
        },
      ],
    ],
    "removeAssigneesFromIssue": Array [
      Array [
        Object {
          "body": Object {
            "assignees": Array [
              "jasonLaster",
            ],
          },
          "number": 4501,
          "owner": "devtools-html",
          "repo": "debugger.html",
        },
      ],
    ],
    "removeLabel": Array [
      Array [
        Object {
          "name": "in progress",
          "number": 4501,
          "owner": "devtools-html",
          "repo": "debugger.html",
        },
      ],
    ],
  },
  "orgs": Object {
    "addTeamMembership": Array [],
    "getTeamMembership": Array [],
  },
}
`;
